front/components/vaults/AddConnectionMenu.tsx

```diff
@@ -0,0 +1,263 @@
+import { Button, Dialog, DropdownMenu, PlusIcon } from "@dust-tt/sparkle";
+import type {
+  ConnectorProvider,
+  ConnectorType,
+  DataSourceType,
+  DataSourceWithConnectorType,
+  PlanType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import {
+  CONNECTOR_PROVIDERS,
+  isConnectorProviderAllowed,
+} from "@dust-tt/types";
+import { useRouter } from "next/router";
+import { useContext, useState } from "react";
+
+import { CreateConnectionConfirmationModal } from "@app/components/data_source/CreateConnectionConfirmationModal";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import {
+  CONNECTOR_CONFIGURATIONS,
+  isConnectorProviderAllowedForPlan,
+} from "@app/lib/connector_providers";
+import type { PostManagedDataSourceRequestBody } from "@app/pages/api/w/[wId]/data_sources/managed";
+import { setupConnection } from "@app/pages/w/[wId]/builder/data-sources/managed";
+
+const REDIRECT_TO_EDIT_PERMISSIONS = [
+  "confluence",
+  "google_drive",
+  "microsoft",
+  "slack",
+  "intercom",
+];
+
+type AddConnectionMenuProps = {
+  owner: WorkspaceType;
+  plan: PlanType;
+  existingDataSources: DataSourceWithConnectorType[];
+  dustClientFacingUrl: string;
+  setIsProviderLoading: (provider: ConnectorProvider, value: boolean) => void;
+};
+
+export const AddConnectionMenu = ({
+  owner,
+  plan,
+  existingDataSources,
+  dustClientFacingUrl,
+  setIsProviderLoading,
+}: AddConnectionMenuProps) => {
+  const sendNotification = useContext(SendNotificationsContext);
+  const [showUpgradePopup, setShowUpgradePopup] = useState<boolean>(false);
+  const [showPreviewPopupForProvider, setShowPreviewPopupForProvider] =
+    useState<{ isOpen: boolean; connector: ConnectorProvider | null }>({
+      isOpen: false,
+      connector: null,
+    });
+  const [showConfirmConnection, setShowConfirmConnection] = useState<{
+    isOpen: boolean;
+    connector: ConnectorProvider | null;
+  }>({
+    isOpen: false,
+    connector: null,
+  });
+
+  const router = useRouter();
+  const allConnectors = CONNECTOR_PROVIDERS.filter(
+    (f) => isConnectorProviderAllowedForPlan(plan, f) && f !== "webcrawler"
```

Comment:
We have a helper for `isWebsite`.

Reply:
isWebsite is on datasource, however i can use isConnectionIdRequiredForProvider here